### PR TITLE
Add 'local authority' column to bookings report

### PR DESCRIPTION
**Pull request by danlivings-dxw**
> See [ticket #1682 on the CAS3 Trello board](https://trello.com/c/plK3ZJVz/1682-add-local-authority-name-into-the-mi-report).

This PR modifies the bookings report to introduce the 'local authority' column, which is derived from the premises. Not all CAS3 premises have a local authority defined, so in these situations, the column will be blank.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt [hunk 1]
```diff
@@ -36,6 +36,7 @@ class BookingsReportGenerator : ReportGenerator<BookingsReportDataAndPersonInfo,
         isReferralEligibleForCas3 = booking.referralEligibleForCas3,
         referralEligibilityReason = booking.referralEligibilityReason,
         probationRegion = booking.probationRegion,
+        localAuthority = booking.localAuthority,
```
Comment (sudhakar-plr) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt:39]: Would it make sense to name it as `localAuthorityAreaName` in the report column?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/repository/BookingsReportRepository.kt [hunk 2]
```diff
@@ -23,6 +23,7 @@ interface BookingsReportRepository : JpaRepository<BookingEntity, UUID> {
       cas3_app.is_eligible AS isReferralEligibleForCas3,
       cas3_app.eligibility_reason AS referralEligibilityReason,
       probation_region.name AS probationRegion,
+      local_authority_area.name AS localAuthority,
```
Comment (sudhakar-plr) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/repository/BookingsReportRepository.kt:26]: Would it make sense to name it as `localAuthorityAreaName`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/repository/BookingsReportRepository.kt [hunk 3]
```diff
@@ -83,6 +86,7 @@ interface BookingsReportData {
   val referralEligibleForCas3: Boolean?
   val referralEligibilityReason: String?
   val probationRegion: String
+  val localAuthority: String?
```
Comment (sudhakar-plr) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/repository/BookingsReportRepository.kt:89]: same as above

**Review by sudhakar-plr — CHANGES_REQUESTED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt (hunk 1 above)
Comment (danlivings-dxw) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt:39]: As these column names are intended to be read by the end user, I don't think `localAuthorityAreaName` communicates any additional information over just `localAuthority`, as the name (e.g. "East Midlands") is what they're going to be referring to it by.

Comment (sudhakar-plr) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BookingsReportGenerator.kt:39]: agree, we can keep it for report for end user.

**Review by tahb — APPROVED** [review]
LGTM 👍

**Review by sudhakar-plr — APPROVED** [review]

**Pull request merged**